tests/integration/test_api_gateway.py

```diff
@@ -722,8 +722,22 @@ def test_put_integration_dynamodb_proxy_validation_with_response_template(self):
         result = table.get_item(Key={'id': 'id1'})
         self.assertEqual(result['Item']['data'], 'foobar123')
 
+    def test_api_key_required_for_methods(self):
+        response_templates = {'application/json': json.dumps({'TableName': 'MusicCollection',
+                                                              'Item': {'id': '$.Id', 'data': '$.data'}})}
+
+        api_id = self.create_api_gateway_and_deploy(response_templates, True)
+        url = self.gateway_request_url(api_id=api_id, stage_name='staging', path='/')
+
+        response = requests.put(
+            url,
+            json.dumps({'id': 'id1', 'data': 'foobar123'}),
+        )
+
+        self.assertEqual(response.status_code, 403)
+
     @staticmethod
-    def create_api_gateway_and_deploy(response_template):
+    def create_api_gateway_and_deploy(response_template, is_api_key_required):
```

Comment:
nitpick: we could change this to:
```
     def create_api_gateway_and_deploy(response_template, is_api_key_required=False):
        ...
```
And then leave lines 698/711 above as:
```
api_id = self.create_api_gateway_and_deploy(response_templates)
```

Reply:
If I do this,other test cases would get impacted.It will return 403 for them.it is the reason why I have added this new variable.